### PR TITLE
feat(style): adding padding margin to style props and theme keys

### PR DESCRIPTION
**Pull request by dbanksdesign**
#### Description of changes

* adding `marginInline`, `marginBlock`, `paddingInline`, `paddingBlock` to style props
* Allowing those, plus `margin` and `padding` to accept a space key

#### Description of how you validated changes

Added unit tests and tried it on the docs site locally. 

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by zchenwei — COMMENTED** [review]
For shortcut, it would be a bit tricky. The function doesn't have the ability to deal with multiple values with space delimiter like `xx xx xx`. Do we still want to add that now?

**Comment by dbanksdesign** [comment]
@zchenwei ah I forgot about the shorthands! Do you think this makes sense? Added some more unit tests

File: packages/react/src/primitives/shared/__tests__/styleUtils.test.tsx [hunk 1]
```diff
@@ -149,6 +149,7 @@ describe('convertStylePropsToStyleObj: ', () => {
     const props: ViewProps = {
       color: 'red',
       fontWeight: 'bold',
+      padding: 'large',
```
Comment (zchenwei) [inline packages/react/src/primitives/shared/__tests__/styleUtils.test.tsx:152]: nit: it seems we don't test the `padding` in this single test. Could simply remove it or add one more `expect` below

File: packages/react/src/primitives/types/style.ts [hunk 2]
```diff
@@ -73,14 +73,18 @@ export interface BaseStyleProps extends FlexItemStyleProps, GridItemStyleProps {
     SpaceKeys<StyleToken<Property.LetterSpacing>>
   >;
   lineHeight?: ResponsiveStyle<LineHeightKeys<StyleToken<Property.LineHeight>>>;
-  margin?: ResponsiveStyle<StyleToken<Property.Margin>>;
+  margin?: ResponsiveStyle<SpaceKeys<StyleToken<Property.Margin>>>;
+  marginBlock?: ResponsiveStyle<SpaceKeys<StyleToken<Property.MarginBlockEnd>>>;
```
Comment (zchenwei) [inline packages/react/src/primitives/types/style.ts:77]:
```suggestion
  marginBlock?: ResponsiveStyle<SpaceKeys<StyleToken<Property.MarginBlock>>>;
```

File: packages/react/src/primitives/types/style.ts [hunk 3]
```diff
@@ -73,14 +73,18 @@ export interface BaseStyleProps extends FlexItemStyleProps, GridItemStyleProps {
...
   marginBlockEnd?: ResponsiveStyle<
     SpaceKeys<StyleToken<Property.MarginBlockEnd>>
   >;
   marginBlockStart?: ResponsiveStyle<
     SpaceKeys<StyleToken<Property.MarginBlockStart>>
   >;
   marginBottom?: ResponsiveStyle<SpaceKeys<StyleToken<Property.MarginBottom>>>;
+  marginInline?: ResponsiveStyle<
+    SpaceKeys<StyleToken<Property.MarginBlockEnd>>
```
Comment (zchenwei) [inline packages/react/src/primitives/types/style.ts:86]:
```suggestion
    SpaceKeys<StyleToken<Property.MarginInline>>
```

File: packages/react/src/primitives/types/style.ts [hunk 4]
```diff
@@ -96,7 +100,8 @@ export interface BaseStyleProps extends FlexItemStyleProps, GridItemStyleProps {
   minWidth?: ResponsiveStyle<SpaceKeys<StyleToken<Property.MinWidth>>>;
   opacity?: ResponsiveStyle<OpacityKeys<StyleToken<Property.Opacity>>>;
   overflow?: ResponsiveStyle<StyleToken<Property.Overflow>>;
-  padding?: ResponsiveStyle<StyleToken<Property.Padding>>;
+  padding?: ResponsiveStyle<SpaceKeys<StyleToken<Property.Padding>>>;
+  paddingBlock?: ResponsiveStyle<SpaceKeys<StyleToken<Property.Padding>>>;
```
Comment (zchenwei) [inline packages/react/src/primitives/types/style.ts:104]:
```suggestion
  paddingBlock?: ResponsiveStyle<SpaceKeys<StyleToken<Property.PaddingBlock>>>;
```

File: packages/react/src/primitives/types/style.ts [hunk 5]
```diff
@@ -106,6 +111,7 @@ export interface BaseStyleProps extends FlexItemStyleProps, GridItemStyleProps {
   paddingBottom?: ResponsiveStyle<
     SpaceKeys<StyleToken<Property.PaddingBottom>>
   >;
+  paddingInline?: ResponsiveStyle<SpaceKeys<StyleToken<Property.Padding>>>;
```
Comment (zchenwei) [inline packages/react/src/primitives/types/style.ts:114]:
```suggestion
  paddingInline?: ResponsiveStyle<SpaceKeys<StyleToken<Property.PaddingInline>>>;
```

**Review by zchenwei — APPROVED** [review]
LGTM! Just a few non blocking comments

**Comment by zchenwei** [comment]
> @zchenwei ah I forgot about the shorthands! Do you think this makes sense? Added some more unit tests

Yeah, perfect now!

**Review by jacoblogan — APPROVED** [review]
LGTM

**Review by reesscot — APPROVED** [review]

**Pull request merged**